File: src/lib/workspaces.ts
import { invoke } from "@tauri-apps/api/core";

export type Workspace = {
  id: string;
  projectPath: string;
  planId?: string;
  branch: string;
  path: string;
  createdAt: number;
  prunedAt?: number;
};

export async function createWorkspace(
  projectPath: string,
  planId: string | null,
  referenceId: string,
  slug: string,
): Promise<Workspace> {
  return invoke<Workspace>("workspace_create", {
    projectPath,
    planId,
    referenceId,
    slug,
  });
}

export async function listWorkspaces(projectPath: string): Promise<Workspace[]> {
  return invoke<Workspace[]>("workspace_list", { projectPath });
}

export async function removeWorkspace(id: string, force: boolean): Promise<void> {
  return invoke<void>("workspace_remove", { id, force });
}

export async function isWorkspaceSupported(projectPath: string): Promise<boolean> {
  return invoke<boolean>("workspace_is_supported", { projectPath });
}

File: src-tauri/src/commands/workspaces.rs
use crate::services::worktree_service::{Workspace, WorktreeService};

#[tauri::command]
pub fn workspace_create(
    project_path: String,
    plan_id: Option<String>,
    reference_id: String,
    slug: String,
) -> Result<Workspace, String> {
    WorktreeService::create(&project_path, plan_id.as_deref(), &reference_id, &slug)
}

#[tauri::command]
pub fn workspace_list(project_path: String) -> Result<Vec<Workspace>, String> {
    WorktreeService::list(&project_path)
}

#[tauri::command]
pub fn workspace_remove(id: String, force: bool) -> Result<(), String> {
    WorktreeService::remove(&id, force)
}

#[tauri::command]
pub fn workspace_is_supported(project_path: String) -> Result<bool, String> {
    Ok(WorktreeService::is_supported(&project_path))
}
